pkg/config/config.go

```diff
@@ -15,9 +24,18 @@ type Target struct {
 	LabelSelector string
 }
 
+// Check is a metrics threshold that should be met to consider a candidate
+// healthy.
+type Check struct {
+	Type       MetricsCheck
+	Percentile float64
+	Max        float64
```

Comment:
What does max do

Reply:
I was following what we currently have in the design doc.

```
strategy:
   ...
   checks:
       - type: request-latency
         percentile: 99
         max: 500
       - type: error-rate
         max: 1
```

Max is the maximum allowed value for each of the health checks. For example, I want the maximum allowed error rate to be 1%.

Reply:
should prob call `error-rate-percent` to make it clear

also how do I represent "min N requests served" in this model?

Reply:
Min is still not supported in this model. I would include it in the PR for request count.

Also, I think the request count should be something that is not a health/metrics check type. For a health check, if the specified condition is not met, a rollback should occur. That's not what we want with the request count. We want to consider the health after we confirmed that the number of requests is what we expect.

If we want to represent the option in YAML format (not a thing we are doing now), I'm thinking of two options.
1)
```
strategy:
   ...
   metrics:
       minRequestCount: 1000
       checks (maybe thresholds?):
           - type: request-latency
             percentile: 99
             max: 500
```

2)

```
strategy:
   steps: [5, 30, 60]
   metricsMinRequestCount: 1000
   metrics:
       - type: request-latency
         percentile: 99
         max: 500
       - type: error-rate
         max: 1
```

Reply:
technically, it's a check. And you have checks[]. why is this not an option:

```
- type: request-count
  min: 1000
```

Reply:
I could do that, but it could be confusing since everything you specified in `metrics` are things that, if not met, should produce a rollback. In the case of the request count, we don't want a rollback because your revision didn't get 1000 requests.

But doing it this way or the ways I showed before is only a concern when dealing with YAML configurations. We could handle that later.

I'll ad a Min field in the metrics check struct since we might add some metrics checks that need it in the future anyways (if request count is not one of them)

Reply:
Valid point. We should've thought this as part of the "design". :) It might be go back to the drawing board at this stage.

Reply:
I'll let this one pass for now. 

We should add probably `minRequestCount` as a check in the future as well, but it would have an implicit "triggersRollback: false" like feature.